### PR TITLE
Store the return value of Fsync for check

### PR DESCRIPTION
**Pull request by riversand963**
Test plan:
```
$make clean && make -j32 all check
```

There is another call of `Fsync` in blob_db, too.

### CODE REVIEW

**Comment by riversand963** [comment]
@yiwu-arbug could you take a look at the change I made to `blob_db_impl.cc`? Is it the correct error handling when fsyncing the dir fails?

File: db/compaction_job.cc [hunk 1]
```diff
@@ -603,7 +604,6 @@ Status CompactionJob::Run() {
   TEST_SYNC_POINT("CompactionJob::Run:BeforeVerify");
 
   // Check if any thread encountered an error during execution
-  Status status;
   for (const auto& state : compact_->sub_compact_states) {
     if (!state.status.ok()) {
       status = state.status;
```
[the remark this answers is not in the document]
Comment (riversand963) [reply db/compaction_job.cc]: Done.

File: utilities/backupable/backupable_db.cc [hunk 2]
```diff
@@ -874,19 +874,19 @@ Status BackupEngineImpl::CreateNewBackupWithMetadata(
         GetAbsolutePath(GetPrivateFileRel(new_backup_id, false)),
         &backup_private_directory);
     if (backup_private_directory != nullptr) {
-      backup_private_directory->Fsync();
+      s = backup_private_directory->Fsync();
     }
     if (private_directory_ != nullptr) {
```
[the remark this answers is not in the document]
Comment (riversand963) [reply utilities/backupable/backupable_db.cc]: Done. Thanks

File: utilities/blob_db/blob_db_impl.cc [hunk 3]
```diff
@@ -1732,7 +1732,10 @@ std::pair<bool, int64_t> BlobDBImpl::DeleteObsoleteFiles(bool aborted) {
 
   // directory change. Fsync
   if (file_deleted) {
-    dir_ent_->Fsync();
+    Status s = dir_ent_->Fsync();
+    if (!s.ok()) {
+      return std::make_pair(false, -1);
```
Comment (yiwu-arbug) [inline utilities/blob_db/blob_db_impl.cc]: I would like to continue execution but print an error log here. BlobDB currently don't expose background error like core rocksdb does.

Comment (riversand963) [reply utilities/blob_db/blob_db_impl.cc]: Done.

**Pull request closed without merging**